File: 202-happy-number/happy-number.cpp
class Solution {
public:
int digSum(int n)
{
    int sum=0;
    int digit;
    while(n>0)
    {
        digit=n%10;
        n=n/10;
        sum=sum+digit*digit;

    }
    return sum;
}
    bool isHappy(int n) {
      int  slow=digSum(n);
        int fast=digSum(digSum(n));
        while(slow!=fast)
        {
            slow=digSum(slow);
            fast=digSum(digSum(fast));
        }

        
        return slow==1;
    }
};

File: 202-happy-number/happy-number.java
class Solution {
     
     

    public int SquareOfDigits(int n)
    {  
        int sum=0;
        while(n>0)
        {
        int dig=n%10;
        sum=sum+dig*dig;
        n=n/10;
        }

        return sum;
    }
    public boolean isHappy(int n) {
        int slow=n;
        int fast=n;

        while(fast!=1)
        {
            slow=SquareOfDigits(slow);
            fast=SquareOfDigits(SquareOfDigits(fast));

            if(fast==1)
            {
                return true;
            }
            if(fast==slow)
            {
                return false;
            }
        }


       return true;

        
    }
}